virttest/utils_test/__init__.py

```diff
@@ -838,205 +848,284 @@ def transfer_data(session, host_cmd, guest_cmd, n_time, timeout,
+        method to clone and install from git repo inside guest
+
+        :param repo_path: repo link to be cloned
         """
-        success = True
 
-        cmd = "[ -n \"$(ls %s)\" ]" % test_path
-        if session.cmd_status(cmd) > 0:
-            cmd = "mkdir -p %s" % test_path
-            session.cmd(cmd)
+        cmd = "cd %s;" % self.test_path
+        cmd += "rm -rf %s" % self.repo_name(repo_path)
+        self.session.cmd(cmd)
+        cmd = "git clone %s;" % repo_path
+        cmd += "cd %s;" % self.repo_name(repo_path)
+        if make:
+            cmd += "make %s;" % make
+        if install:
+            cmd += "%s setup.py install" % self.python
+        return self.session.cmd_status(cmd) == 0
 
-        cmd = "[ -n \"$(ls %s)\" ]" % result_path
-        if session.cmd_status(cmd) > 0:
-            cmd = "mkdir -p %s" % result_path
-            session.cmd(cmd)
+    def repo_name(self, repo_path):
+        """ Wrapper to return the repo name """
+        return repo_path.split(os.sep)[-1].split(".")[0]
 
+    def runtest(self):
+        """
+        Run test method to download the tests and trigger avocado command
+        """
         logging.debug("Downloading Test")
-        cmd = "cd %s;[ -d %s ] || git clone %s" % (test_path,
-                                                   testrepo.split("/")[-1].split('.')[0],
-                                                   testrepo)
-        status, output = session.cmd_status_output(cmd, timeout=100)
-        if status > 0:
-            raise exceptions.TestError("Downloading test failed: %s" % output)
+        if self.avocado_vt:
+            cmd = "avocado vt-bootstrap --vt-type %s" % self.vt_type
+            if self.guest_image:
+                cmd += " --vt-guest-os %s --yes-to-all" % self.guest_image
+            status, output = self.session.cmd_status_output(cmd, timeout=100)
+            if status != 0:
+                raise exceptions.TestError("Downloading test failed: %s" % output)
+        else:
+            if not self.git_install(self.test_repo, install=False):
+                raise exceptions.TestError("Downloading test failed")
+
         logging.debug("Running Test")
-        for test in testlist:
-            mux = ""
-            testcase = test[0]
-            testcase = os.path.join(test_path,
-                                    testrepo.split("/")[-1].split('.')[0],
-                                    testcase)
-            try:
-                mux = test[1]
-            except KeyError:
-                pass
-            cmd = "avocado run %s --job-results-dir %s --job-timeout %d" % (
-                testcase, result_path, timeout)
-            if mux:
-                cmd += " -m %s" % mux
-            if add_args:
-                cmd += " %s" % add_args
-            status, output = session.cmd_status_output(cmd, timeout=timeout)
-            if status > 0:
+        cmd = "avocado run %s"
+        for test_each in self.testlist:
```

Comment:
for avocado-vt at this moment there is no test specific filters or args passed unlike mux file in misc tests so we need not iterate and run avocado run multiple times, rather we can run all tests together...make it simple at present..

Reply:
I understand it might helpful to report per test status, but avocado-vt already have a good per-test view even we run multiple tests together....